block-storage/src/main/scala/coop/rchain/blockstorage/FileLMDBIndexBlockStore.scala

```diff
@@ -87,25 +89,13 @@ class FileLMDBIndexBlockStore[F[_]: Monad: Sync: Log] private (
     state.modify(s => s.copy(currentIndex = f(s.currentIndex)))
 
   private def readBlockMessage(indexEntry: IndexEntry): StorageIOErrT[F, BlockMessage] = {
-    def readBlockMessageFromFile(storageFile: RandomAccessFile): StorageIOErrT[F, BlockMessage] =
+    def readBlockMessageFromFile(storageFile: RandomAccessIO[F]): StorageIOErrT[F, BlockMessage] =
       for {
-        _ <- EitherT(Sync[F].delay { storageFile.seek(indexEntry.offset) }.attempt)
-              .leftMap[StorageIOError] {
-                case e: IOException => FileSeekFailed(e)
-                case e              => UnexpectedIOStorageError(e)
-              }
-        blockMessageSize <- EitherT(Sync[F].delay { storageFile.readInt() }.attempt)
-                             .leftMap[StorageIOError] {
-                               case e: IOException => IntReadFailed(e)
-                               case e              => UnexpectedIOStorageError(e)
-                             }
+        _                      <- toStorageIOErrT(storageFile.seek(indexEntry.offset))
+        blockMessageSize       <- toStorageIOErrT(storageFile.readInt)
         blockMessagesByteArray = Array.ofDim[Byte](blockMessageSize)
-        _ <- EitherT(Sync[F].delay { storageFile.readFully(blockMessagesByteArray) }.attempt)
-              .leftMap[StorageIOError] {
-                case e: IOException => ByteArrayReadFailed(e)
-                case e              => UnexpectedIOStorageError(e)
-              }
-        blockMessage = BlockMessage.parseFrom(blockMessagesByteArray)
+        _                      <- toStorageIOErrT(storageFile.readFully(blockMessagesByteArray))
+        blockMessage           = BlockMessage.parseFrom(blockMessagesByteArray)
```

Comment:
This is a **very** neat refactoring. I love how IDEA shows how boilerplate and noise is disappearing around the actual method calls that remain unchanged :heart:

This made me less cautious though and I didn't check that the boilerplate was faithfully moved to the pure wrapper. It doesn't matter though, as we rarely (never?) actually handle those errors other than by terminating execution. So even if it ends up being an UnexpectedError - it's fine.

Good job! :tada: